File: src/main/java/ReverseLinkedList/GoKu-gaga/ReverseLinkedList.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
  let pre = null
  while (head) {
      next = head.next
      head.next = pre
      pre = head
      head = next
  }
  return pre
};

/** 
 * 执行结果
 * 执行用时 : 80 ms, 在所有JavaScript提交中击败了 94.76% 的用户
 * 内存消耗 : 35.3 MB, 在所有JavaScript提交中击败了 21.07% 的用户
*/

File: src/main/java/ReverseLinkedList/zhangwentao/ReverseLinkedList.py
#!/usr/bin/python
# -*- coding: UTF-8 -*-
#!@Author:zhangwentao

# Definition for singly-linked list.
class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution(object):
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        todo = None
        now = head
        pre = None;
        while now != None:
            todo = now.next;
            now.next = pre;
            pre = now;
            now = todo;
        return pre;
